lib/pacemaker/pcmk_output.c

```diff
@@ -1864,6 +1864,50 @@ attribute_xml(pcmk__output_t *out, va_list args)
     return pcmk_rc_ok;
 }
 
+PCMK__OUTPUT_ARGS("rule-check", "const char *", "int")
+static int
+rule_check_default(pcmk__output_t *out, va_list args)
+{
+    const char *rule_id = va_arg(args, const char *);
+    int result = va_arg(args, int);
+
+    if (result == pcmk_rc_within_range) {
+        out->info(out, "Rule %s is still in effect", rule_id);
+    } else if (result == pcmk_rc_ok) {
+        out->info(out, "Rule %s satisfies conditions", rule_id);
+    } else if (result == pcmk_rc_after_range) {
+        out->info(out, "Rule %s is expired", rule_id);
+    } else if (result == pcmk_rc_before_range) {
+        out->info(out, "Rule %s has not yet taken effect", rule_id);
+    } else if (result == pcmk_rc_op_unsatisfied) {
+        out->info(out, "Rule %s does not satisfy conditions", rule_id);
+    } else {
+        out->info(out, "Could not determine whether rule %s is expired",
+                  rule_id);
+    }
+
+    return pcmk_rc_ok;
+}
+
+PCMK__OUTPUT_ARGS("rule-check", "const char *", "int")
+static int
+rule_check_xml(pcmk__output_t *out, va_list args)
+{
+    const char *rule_id = va_arg(args, const char *);
+    int result = va_arg(args, int);
+
+    char *rc_str = pcmk__itoa(pcmk_rc2exitc(result));
+
+    pcmk__output_create_xml_node(out, "rule-check",
+                                 "rule-id", rule_id,
+                                 "rc", rc_str,
+                                 NULL);
```

Comment:
The messages from rule_check_default could be added to the XML for additional clarity. Not sure if that's desirable and whether that would be allowable in the context of schema compatibility.

Reply:
Not worth the effort at this point

Reply:
With the latest update, this seems trivial to implement if we wanted to do so. We could add an optional `message` attribute and set it to a known message or something like `"Error occurred checking rule"`. It could go in a separate PR if we wanted it, since this one is already unwieldy.